components/director/internal/domain/eventdef/converter.go

```diff
@@ -188,7 +195,7 @@ func (c *converter) ToEntity(eventModel *model.EventDefinition) *Entity {
 		Successors:          repo.NewNullableStringFromJSONRawMessage(eventModel.Successors),
 		ChangeLogEntries:    repo.NewNullableStringFromJSONRawMessage(eventModel.ChangeLogEntries),
 		Labels:              repo.NewNullableStringFromJSONRawMessage(eventModel.Labels),
-		Visibility:          repo.NewNullableString(eventModel.Visibility),
+		Visibility:          repo.NewNullableString(visibility),
```

Comment:
You can make the visibility column string instead of NullableString and add not null constraint in the database in order to ensure that the if above will never be missed.

Valid for the API Entity as well.